pkg/kube/inject/webhook.go

```diff
@@ -1075,13 +1078,16 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 		injectedAnnotations: wh.Config.InjectedAnnotations,
 		proxyEnvs:           parseInjectEnvs(path),
 	}
-	clusterID, _ := extractClusterAndNetwork(params)
-	if wh.namespaces != nil {
-		client := wh.namespaces.ForCluster(cluster.ID(clusterID))
+
+	if openShiftNamespaceTest != nil {
+		// Mocked namespace to cover unit tests
+		params.namespace = openShiftNamespaceTest
+	} else if wh.namespaces != nil {
+		client := wh.namespaces.ForCluster(wh.clusterID)
```

Comment:
I don't get it. This is supposed to look up the clusterID _of the request_. Now it seems like its always looking up the same cluster, which means its basically not multi-cluster.


I think in https://github.com/istio/istio/blob/05d0d5a529e1e9f83736b78b1808ba122536ec2a/pkg/kube/multicluster/component.go#L42 we can just add some logic so that if `""` is passed, we default it to `configClusterID` (which we will need to pass through into the Component, but should be simple)